app/routes/_layout+/me.assets.tsx

```diff
@@ -0,0 +1,107 @@
+import { json, redirect, type LoaderFunctionArgs } from "@remix-run/node";
+import { getPaginatedAndFilterableAssets } from "~/modules/asset/service.server";
+import {
+  getFiltersFromRequest,
+  setCookie,
+  userPrefs,
+} from "~/utils/cookies.server";
+import { makeShelfError } from "~/utils/error";
+import { data, error } from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.data";
+import { requirePermission } from "~/utils/roles.server";
+import { AssetsList } from "./assets._index";
+
+export async function loader({ request, context }: LoaderFunctionArgs) {
+  const authSession = context.getSession();
+  const userId = authSession.userId;
+
+  try {
+    const { organizationId } = await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.asset,
+      action: PermissionAction.read,
+    });
+
+    const { filters, redirectNeeded } = await getFiltersFromRequest(
```

Comment:
From line 29 to the end of the loader is the same as the team Member loader. I think this should be the case for all those routes. Shall we create functions to do those things so we dont repeat the code in those pages and teamMember pages.